homeassistant/components/iqvia/__init__.py

```diff
@@ -245,13 +298,33 @@ def unit_of_measurement(self):
         @callback
         def update():
             """Update the state."""
-            self.async_schedule_update_ha_state(True)
+            self.update_from_latest_data()
+            self.async_write_ha_state()
 
         self._async_unsub_dispatcher_connect = async_dispatcher_connect(
             self.hass, TOPIC_DATA_UPDATE, update
         )
 
+        await self._iqvia.async_register_api_interest(self._type)
+        if self._type == TYPE_ALLERGY_FORECAST:
+            # Entities that express interest in allergy forecast data should also
+            # express interest in allergy outlook data:
+            await self._iqvia.async_register_api_interest(TYPE_ALLERGY_OUTLOOK)
+
+        update()
```

Comment:
Home Assistant will write it to the state machine after calling `async_added_to_hass` so this is enough:

```suggestion
        self.update_from_latest_data()
```